src/core/function_admin.tf

```
@@ -0,0 +1,407 @@
+      SERVICE_PRINCIPAL_SECRET    = data.azurerm_key_vault_secret.ad_APPCLIENT_APIM_SECRET.value
+      SERVICE_PRINCIPAL_TENANT_ID = data.azurerm_key_vault_secret.common_AZURE_TENANT_ID.value
+
+      PUBLIC_API_KEY = data.azurerm_key_vault_secret.apim_IO_GDPR_SERVICE_KEY.value
+
+      SESSION_API_KEY = data.azurerm_key_vault_secret.app_backend_PRE_SHARED_KEY.value
+
+      __DISABLED__SENDGRID_API_KEY = data.azurerm_key_vault_secret.common_SENDGRID_APIKEY.value
+      MAILUP_USERNAME              = data.azurerm_key_vault_secret.common_MAILUP_USERNAME.value
+      MAILUP_SECRET                = data.azurerm_key_vault_secret.common_MAILUP_SECRET.value
+    }
+  }
+}
+
+# Admin resouce group
+resource "azurerm_resource_group" "admin_rg" {
+  name     = format("%s-admin-rg", local.project)
+  location = var.location
+
+  tags = var.tags
+}
+
+# Subnet to host admin function
+module "admin_snet" {
+  source                                         = "git::https://github.com/pagopa/azurerm.git//subnet?ref=v1.0.51"
+  name                                           = format("%s-admin-snet", local.project)
+  address_prefixes                               = var.cidr_subnet_fnadmin
+  resource_group_name                            = data.azurerm_resource_group.vnet_common_rg.name
+  virtual_network_name                           = data.azurerm_virtual_network.vnet_common.name
+  enforce_private_link_endpoint_network_policies = true
+
+  service_endpoints = [
+    "Microsoft.Web",
+    "Microsoft.AzureCosmosDB",
+    "Microsoft.Storage",
+  ]
+
+  delegation = {
+    name = "default"
+    service_delegation = {
+      name    = "Microsoft.Web/serverFarms"
+      actions = ["Microsoft.Network/virtualNetworks/subnets/action"]
+    }
+  }
+}
+
+module "function_admin" {
+  source = "git::https://github.com/pagopa/azurerm.git//function_app?ref=v3.4.0"
+
+  resource_group_name = azurerm_resource_group.admin_rg.name
+  name                = format("%s-admin-fn", local.project)
+  location            = var.location
+  health_check_path   = "/info"
+
+  os_type          = "linux"
+  linux_fx_version = "NODE|14"
+  runtime_version  = "~4"
+
+  always_on                                = "true"
+  application_insights_instrumentation_key = data.azurerm_application_insights.application_insights.instrumentation_key
+
+  app_service_plan_info = {
+    kind                         = var.function_admin_kind
+    sku_tier                     = var.function_admin_sku_tier
+    sku_size                     = var.function_admin_sku_size
+    maximum_elastic_worker_count = 0
+  }
+
+  app_settings = merge(
+    local.function_admin.app_settings_common, {
+      # TODO: All of this can be removed?
+      "AzureWebJobs.UpdateVisibleServicesCache.Disabled"             = "1"
+      "AzureWebJobs.UpdateVisibleServicesCacheOrchestrator.Disabled" = "1"
+      "AzureWebJobs.UpdateVisibleServicesCacheActivity.Disabled"     = "1"
```

```suggestion
      # Disabled CosmosDB Trigger Activity on slot
      "AzureWebJobs.UserDataProcessingTrigger.Disabled" = "0",
```
we need to force to enable the function on prod slot

I think that we can remove this settings because the functions does not exists

UpdateVisibleServicesCache
UpdateVisibleServicesCacheOrchestrator
UpdateVisibleServicesCacheActivity

---

@BurnedMarshal ping :)

---

Fixed in [b253b32](https://github.com/pagopa/io-infra/pull/315/commits/b253b329c7952d66d809818d88378280082e96d6)